File: java/Snackbar/app/src/main/java/com/github/shellhub/snackbar/MainActivity.java
package com.github.shellhub.snackbar;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.send:
                Snackbar.make(view, R.string.delete_msg, Snackbar.LENGTH_LONG)
                        .setAction(R.string.undo, new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                Toast.makeText(MainActivity.this, "Message recovery", Toast.LENGTH_SHORT).show();
                            }
                        }).show();
                break;
            default:
                break;
        }
    }
}

File: kotlin/Snackbar/app/src/main/java/com/github/shellhub/snackbar/MainActivity.kt
package com.github.shellhub.snackbar

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    
    /* execute when view clicked */
    fun onClick(view: View) {
        when (view.id) {
            R.id.send -> Snackbar.make(
                view,
                R.string.delete_msg,
                Snackbar.LENGTH_LONG
            ).setAction(
                R.string.undo
            ) {
                Toast.makeText(
                    this,
                    R.string.recovery_msg,
                    Toast.LENGTH_LONG
                ).show()
            }.show();
        }
    }
}
